### PR TITLE
[Next12-ChakraUI kit] adding jest testing

### LINKED ISSUE #326: [Next-Chakra UI] Add Jest tests
Acceptance criteria: 

- install Jest
- add Jest test for Counter example
- add Jest test for fetch example

### PR DESCRIPTION
**Pull request by jdwilkin4**
## Type of change

- [ ] Documentation change
- [ ] Bug fix
- [x] Testing

## Summary of change
This PR adds Jest testing to the Next12 and Chakra UI starter kit. 

## Checklist

- [x] These changes follow the [contributing guidelines](https://github.com/thisdot/starter.dev/blob/main/CONTRIBUTING.md)
- [x] This starter kit has been approved by the maintainers
- [x] Changes for this new starter kit are being pushed to an integration branch instead of main
- [x] All dependencies are version locked
- [x] This fix resolves #326 
- [x] I have verified the fix works and introduces no further errors

### CODE REVIEW

File: starters/next12-routing-chakra-ui/__tests__/counter-example.test.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import CounterExample from "../src/pages/counter-example/index";
+import "@testing-library/jest-dom";
```
Comment (danecando) [inline starters/next12-routing-chakra-ui/__tests__/counter-example.test.tsx:3]: Is this needed to be imported in the test file? I thought the testEnvironment option in the jest config handled that?

File: starters/next12-routing-chakra-ui/__tests__/fetch-example.test.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import fetch from "jest-fetch-mock";
+import { render, screen } from "@testing-library/react";
+import FetchExample from "../src/pages/fetch-example/[[...slug]]";
+import "@testing-library/jest-dom";
+
+describe("Counter Example page", () => {
```
Comment (danecando) [inline starters/next12-routing-chakra-ui/__tests__/fetch-example.test.tsx:6]: Probably want to rename this one

**Review by danecando — CHANGES_REQUESTED** [review]
Just one question and one small thing that I noticed to update

File: starters/next12-routing-chakra-ui/__tests__/counter-example.test.tsx (hunk 1 above)
Comment (jdwilkin4) [reply starters/next12-routing-chakra-ui/__tests__/counter-example.test.tsx:3]: If I try to remove that import, then 4 tests will fail. 
Screen Shot 2022-12-09 at 12 54 49 PM

Screen Shot 2022-12-09 at 12 55 11 PM

File: starters/next12-routing-chakra-ui/__tests__/fetch-example.test.tsx (hunk 2 above)
Comment (jdwilkin4) [reply starters/next12-routing-chakra-ui/__tests__/fetch-example.test.tsx:6]: Oh, oopps.... good catch

**Review by danecando — APPROVED** [review]

**Pull request merged**